Building amazon instance creates an AMI, but errors

Building amazon instance creates an AMI, but errors:

2017/01/04 15:25:06 ui error: 
==> Some builds didn't complete successfully and had errors:
2017/01/04 15:25:06 machine readable: amazon-instance,error []string{"unexpected EOF"}
2017/01/04 15:25:06 ui error: --> amazon-instance: unexpected EOF

-Packer v0.12.1
-Ubuntu 16
- https://gist.github.com/jsheflin/6fe795727b3fac4c82eaaea46ad695c2

Here is the crash log when running from `master`:

```
panic: interface conversion: interface {} is nil, not map[string][]string
2017/01/06 08:14:31 packer: 
2017/01/06 08:14:31 packer: goroutine 27 [running]:
2017/01/06 08:14:31 packer: panic(0x192b840, 0xc82030fb00)
2017/01/06 08:14:31 packer:     /usr/lib/go-1.6/src/runtime/panic.go:481 +0x3e6
2017/01/06 08:14:31 packer: github.com/mitchellh/packer/builder/amazon/common.(*StepAMIRegionCopy).Run(0xc8204348d0, 0x7fcbf02b4580, 0xc820434840, 0x0)
2017/01/06 08:14:31 packer:     /REDACTED/gopath/src/github.com/mitchellh/packer/builder/amazon/common/step_ami_region_copy.go:26 +0xab2
2017/01/06 08:14:31 packer: github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep.(*BasicRunner).Run(0xc8204365c0, 0x7fcbf02b4580, 0xc820434840)
2017/01/06 08:14:31 packer:     /REDACTED/gopath/src/github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep/basic_runner.go:70 +0x330
2017/01/06 08:14:31 packer: github.com/mitchellh/packer/builder/amazon/instance.(*Builder).Run(0xc820220580, 0x7fcbf02b3fd8, 0xc820402400, 0x7fcbf02b4020, 0xc820408810, 0x7fcbf02b4050, 0xc8202cc268, 0x0, 0x0, 0x0, ...)
2017/01/06 08:14:31 packer:     /REDACTED/gopath/src/github.com/mitchellh/packer/builder/amazon/instance/builder.go:277 +0x1bba
2017/01/06 08:14:31 packer: github.com/mitchellh/packer/packer/rpc.(*BuilderServer).Run(0xc8202824c0, 0x1, 0xc820201be0, 0x0, 0x0)
2017/01/06 08:14:31 packer:     /REDACTED/gopath/src/github.com/mitchellh/packer/packer/rpc/builder.go:93 +0x3b9
2017/01/06 08:14:31 packer: reflect.Value.call(0x17b6500, 0x19a6a18, 0x13, 0x1c54028, 0x4, 0xc8205b9ec8, 0x3, 0x3, 0x0, 0x0, ...)
2017/01/06 08:14:31 packer:     /usr/lib/go-1.6/src/reflect/value.go:435 +0x120d
2017/01/06 08:14:31 packer: reflect.Value.Call(0x17b6500, 0x19a6a18, 0x13, 0xc8205b9ec8, 0x3, 0x3, 0x0, 0x0, 0x0)
2017/01/06 08:14:31 packer:     /usr/lib/go-1.6/src/reflect/value.go:303 +0xb1
2017/01/06 08:14:31 packer: net/rpc.(*service).call(0xc8202940c0, 0xc820294080, 0xc820296090, 0xc820151880, 0xc8202b0560, 0x156f760, 0xc820201bdc, 0x18a, 0x152f140, 0xc820201be0, ...)
2017/01/06 08:14:31 packer:     /usr/lib/go-1.6/src/net/rpc/server.go:383 +0x1c2
2017/01/06 08:14:31 packer: created by net/rpc.(*Server).ServeCodec
2017/01/06 08:14:31 packer:     /usr/lib/go-1.6/src/net/rpc/server.go:477 +0x49d
2017/01/06 08:14:31 /REDACTED/gopath/bin/packer: plugin process exited
2017/01/06 08:14:31 ui error: Build 'hvm.instance-store' errored: unexpected EOF
2017/01/06 08:14:31 Builds completed. Waiting on interrupt barrier...
2017/01/06 08:14:31 machine readable: error-count []string{"1"}
2017/01/06 08:14:31 ui error: 
==> Some builds didn't complete successfully and had errors:
2017/01/06 08:14:31 machine readable: hvm.instance-store,error []string{"unexpected EOF"}
2017/01/06 08:14:31 ui error: --> hvm.instance-store: unexpected EOF
2017/01/06 08:14:31 ui: 
==> Builds finished but no artifacts were created.
2017/01/06 08:14:31 waiting for all plugin processes to complete...
2017/01/06 08:14:31 /REDACTED/gopath/bin/packer: plugin process exited
2017/01/06 08:14:31 /REDACTED/gopath/bin/packer: plugin process exited
2017/01/06 08:14:31 /REDACTED/gopath/bin/packer: plugin process exited
```

--- a/builder/amazon/instance/step_register_ami.go
+++ b/builder/amazon/instance/step_register_ami.go
@@ -63,6 +63,8 @@ func (s *StepRegisterAMI) Run(state multistep.StateBag) multistep.StepAction {
 		return multistep.ActionHalt
 	}
 
+	state.Put("snapshots", map[string][]string{})
+
 	return multistep.ActionContinue
 }
 

